File: Estrutura_Condicional/C/Ex1828/main.c
#include <stdio.h>
#include <string.h>

int main() {
    int T;
    scanf("%d", &T);

    char *regras[][2] = {
        {"tesoura", "papel"},
        {"papel", "pedra"},
        {"pedra", "lagarto"},
        {"lagarto", "Spock"},
        {"Spock", "tesoura"},
        {"tesoura", "lagarto"},
        {"lagarto", "papel"},
        {"papel", "Spock"},
        {"Spock", "pedra"},
        {"pedra", "tesoura"},
        {"tesoura", "Spock"},
        {"Spock", "papel"},
        {"papel", "tesoura"},
        {"tesoura", "pedra"},
        {"pedra", "papel"},
        {"papel", "lagarto"},
        {"lagarto", "pedra"},
        {"pedra", "Spock"},
        {"Spock", "lagarto"},
        {"lagarto", "tesoura"}
    };

    char escolha_sheldon[10], escolha_raj[10];
    for (int t = 1; t <= T; t++) {
        scanf("%s %s", escolha_sheldon, escolha_raj);
        if (strcmp(escolha_sheldon, escolha_raj) == 0) {
            printf("Caso #%d: De novo!\n", t);
        } else {
            char *resultado = NULL;
            for (int i = 0; i < sizeof(regras) / sizeof(regras[0]); i++) {
                if (strcmp(regras[i][0], escolha_sheldon) == 0 && strcmp(regras[i][1], escolha_raj) == 0) {
                    resultado = "Bazinga!";
                    break;
                } else if (strcmp(regras[i][0], escolha_raj) == 0 && strcmp(regras[i][1], escolha_sheldon) == 0) {
                    resultado = "Raj trapaceou!";
                    break;
                }
            }
            printf("Caso #%d: %s\n", t, resultado);
        }
    }

    return 0;
}

File: Estrutura_Condicional/C#/Ex1828/Main.cs
using System;

namespace Ex1828
{
    class Program
    {
        static void Main(string[] args)
        {
            int T = int.Parse(Console.ReadLine());

            string[][] regras = new string[][] {
                new string[] {"tesoura", "papel"},
                new string[] {"papel", "pedra"},
                new string[] {"pedra", "lagarto"},
                new string[] {"lagarto", "Spock"},
                new string[] {"Spock", "tesoura"},
                new string[] {"tesoura", "lagarto"},
                new string[] {"lagarto", "papel"},
                new string[] {"papel", "Spock"},
                new string[] {"Spock", "pedra"},
                new string[] {"pedra", "tesoura"},
                new string[] {"tesoura", "Spock"},
                new string[] {"Spock", "papel"},
                new string[] {"papel", "tesoura"},
                new string[] {"tesoura", "pedra"},
                new string[] {"pedra", "papel"},
                new string[] {"papel", "lagarto"},
                new string[] {"lagarto", "pedra"},
                new string[] {"pedra", "Spock"},
                new string[] {"Spock", "lagarto"},
                new string[] {"lagarto", "tesoura"}
            };

            for (int t = 1; t <= T; t++)
            {
                string[] escolhas = Console.ReadLine().Split(' ');
                string escolhaSheldon = escolhas[0];
                string escolhaRaj = escolhas[1];

                if (escolhaSheldon == escolhaRaj)
                {
                    Console.WriteLine($"Caso #{t}: De novo!");
                }
                else
                {
                    string resultado = null;
                    foreach (string[] regra in regras)
                    {
                        if (regra[0] == escolhaSheldon && regra[1] == escolhaRaj)
                        {
                            resultado = "Bazinga!";
                            break;
                        }
                        else if (regra[0] == escolhaRaj && regra[1] == escolhaSheldon)
                        {
                            resultado = "Raj trapaceou!";
                            break;
                        }
                    }
                    Console.WriteLine($"Caso #{t}: {resultado}");
                }
            }
        }
    }
}